### PR TITLE
LTD-4143 order generated_documents results by created_at

### PR DESCRIPTION
**Pull request by Tllew**
### Order generated documents by created_at instead of name
this resolves some of the issues we're having with ordering of the tables.

[LTD-4143](https://uktrade.atlassian.net/browse/LTD-4143)

[LTD-4143]: https://uktrade.atlassian.net/browse/LTD-4143?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/cases/views/views.py [hunk 1]
```diff
@@ -411,16 +411,29 @@ def get(self, request, pk):
...
+            advice_documents[key]["document"] = document
+
+        return JsonResponse(data={"documents": advice_documents}, status=status.HTTP_200_OK)
+
+    def get_latest_generated_case_documents(self, final_advice, pk):
         # Get other decision documents
-        generated_advice_documents = GeneratedCaseDocument.objects.filter(advice_type__in=final_advice, case__id=pk)
+        generated_advice_documents = GeneratedCaseDocument.objects.filter(
+            advice_type__in=final_advice, case__id=pk
+        ).order_by("created_at")
```
Comment (saruniitr) [inline api/cases/views/views.py:426]: I am not sure if this is applicable in case of regenerating documents, in that case the ordering won't be correct.

Comment (Tllew) [reply api/cases/views/views.py]: marking this as resolved, we had a conversation on slack to clear it up

File: api/cases/views/views.py [hunk 2]
```diff
@@ -411,16 +411,29 @@ def get(self, request, pk):
...
             many=True,
         ).data
-        for document in generated_advice_documents:
-            advice_documents[document["advice_type"]["key"]]["document"] = document
 
-        return JsonResponse(data={"documents": advice_documents}, status=status.HTTP_200_OK)
+        # Remove duplicates for each advice type and filters it to only the most recent advice
+        latest_documents = {}
+        for document in generated_advice_documents:
+            latest_documents[document["advice_type"]["key"]] = document
```
[the remark this answers is not in the document]
Comment (Tllew) [reply api/cases/views/views.py]: agreed, looks nicer and more explicit

**Review by saruniitr — APPROVED** [review]

File: api/cases/views/views.py [hunk 3]
```diff
@@ -411,17 +411,33 @@ def get(self, request, pk):
...
+        # Get other decision documents
+        latest_documents = {}
+        for advice_type in final_advice:
+            # Remove duplicates for each advice type and filters it to only the most recent advice
+            document = self.get_case_document(advice_type, pk)
+            if document:
+                serialised_document = AdviceDocumentGovSerializer(
+                    document,
+                ).data
+                if serialised_document:
```
Comment (currycoder) [inline api/cases/views/views.py:437]: I think if you bin this line codecov will pass - I don't think it's necessary anyways

**Pull request merged**